system-addon/test/functional/mochitest/browser_highlights_section.js

```diff
@@ -1,47 +1,13 @@
 "use strict";
 
-registerCleanupFunction(async () => {
-  await clearHistoryAndBookmarks();
-});
-
-function refreshHighlightsFeed() {
-  // Toggling the pref will clear the feed cache and force a places query.
-  Services.prefs.setBoolPref("browser.newtabpage.activity-stream.feeds.section.highlights", false);
-  Services.prefs.setBoolPref("browser.newtabpage.activity-stream.feeds.section.highlights", true);
-}
-
-async function add_highlights_bookmark(count) {
-  const bookmarks = new Array(count).fill(null).map((entry, i) => ({
-    parentGuid: PlacesUtils.bookmarks.unfiledGuid,
-    title: "foo",
-    url: `https://mozilla${i}.com/nowNew`
-  }));
-
-  for (let placeInfo of bookmarks) {
-    await PlacesUtils.bookmarks.insert(placeInfo);
-  }
-
-  // Add bookmark visits
-  for (let placeInfo of bookmarks) {
-    await PlacesTestUtils.addVisits(placeInfo.url);
-  }
-
-  // Force HighlightsFeed to make a request for the new items.
-  refreshHighlightsFeed();
-}
-
-async function addHighlightsCards(browser, count) {
-  await clearHistoryAndBookmarks();
-  await add_highlights_bookmark(count);
-  await ContentTask.spawn(browser, null, async () => {
-    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".card-outer:not(.placeholder)"),
-      "No highlights cards found.");
-  });
-}
-
 test_newtab({
   async before({tab}) {
-    await addHighlightsCards(tab.linkedBrowser, 2);
+    await addHighlightsBookmarks(2);
+    // Wait for HighlightsFeed to update and display the items.
+    await ContentTask.spawn(tab.linkedBrowser, null, async () => {
+      await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".card-outer:not(.placeholder)"),
+        "No highlights cards found.");
```

Comment:
Did the `test_highlights` approach not work?

Reply:
I misread your comment. So just keep 1 generic test_highlights and pass in additional assertions in a fn.